routers/repo/pull.go

```diff
@@ -420,19 +418,13 @@ func PrepareViewPullInfo(ctx *context.Context, issue *models.Issue) *git.Compare
 	}
 
 	if headBranchExist {
-		allowUpdate, err := pull_service.IsUserAllowedToUpdate(pull, ctx.User)
-		if err != nil {
-			ctx.ServerError("IsUserAllowedToUpdate", err)
-			return nil
-		}
-		ctx.Data["UpdateAllowed"] = allowUpdate
-
-		divergence, err := pull_service.GetDiverging(pull)
-		if err != nil {
-			ctx.ServerError("GetDiverging", err)
-			return nil
+		if pull.CommitsBehind > 0 {
```

Comment:
Why we change the logic since the title said we will do some cache thing?

Reply:
@lunny this is a smal **refactor** only posible because of caching
-> why should we check if Update is Allowed if nothing can be Updated?